### ISSUE TITLE
stylefmt-loader causes an infinite loop with dev-server

### ISSUE THREAD

**Issue by indfnzo** [tomasAlabes/stylefmt-loader#2]
Given the `webpack.config.js` below, stylefmt-loader causes webpack-dev-server to loop indefinitely. stylefmt modifies the scss source, and since the source has been modified, webpack rebuilds everything all over again, rerunning stylefmt again, and causing the infinite loop.

Is there any workaround or fix for this?

``` js
{
    test: /\.scss$/,
    include: [ path.resolve(__dirname, "src/styles") ],
    loaders: [ "style", "css", "postcss", "sass", "stylefmt?config=.stylelintrc.js" ]
}
```

**Comment by tomasAlabes** [comment]
But after running stylefmt the 2nd time the file shouldn't change anymore, am I right? As it fixed all it could the 1st time.

**Comment by indfnzo** [comment]
Nope, it keeps running in an infinite loop. I haven't figured out the cause for it, but it keeps running even if stylefmt isn't actually changing any source file.

**Comment by tomasAlabes** [comment]
It would be very helpful if you can create a minimal repo were this can be reproduced.

**Comment by digital-wonderland** [comment]
I'm running into the same issue. For an example please check out the ```stylefmt-loader-infinite-loop``` branch within the [digital-wonderland/play-webpack-typescript-react](https://github.com/digital-wonderland/play-webpack-typescript-react) repo (all the scala stuff can be safely ignored).

Then run ```npm install``` and ```npm start```. ```webpack-dev-server``` will go into an infinite loop, which happens regardless if there is an error within the scss files or not.

Perhaps having a look at [wbuchwalter/tslint-loader](https://github.com/wbuchwalter/tslint-loader) or [joekallen/tsfmt-loader](https://github.com/joekallen/tsfmt-loader) helps since those work as expected (they halt the build until a lint issue is fixed or reformat the code during build without interruption). I don't know much about building webpack plugins but noteworthy might be that both get registered as a ```pre``` loader?

**Comment by indfnzo** [comment]
@digital-wonderland thanks a lot for creating a sample repository.

I haven't had enough time getting into making one so that's a real time saver.

**Comment by tomasAlabes** [comment]
I can reproduce the issue but I can't find the problem yet.  Even if I use `stylefmt-loader` as a pre-loader. 
The [tsfmt-loader](https://github.com/joekallen/tsfmt-loader/blob/master/index.js) is almost identical (I tried changing some things as `this.cacheable()` or using `writeFileSync` but had no luck.
It must happen when it's mixed with other loaders. @jemhuntr did it happen to you also with typescript loaders?

**Comment by shai32** [comment]
I have the exact same probelm

**Comment by kenisteward** [comment]
@tomasAlabes Created PR #8 to fix this issue.  Would be very happy if you could merge this asap so I can integrate this into my build.

Basically the idea was that since you write a file no matter if it is different or not for every file each write causes Webpack to check and optimize.

**Comment by tomasAlabes** [comment]
Merged in v1.2.3 and verified with the branch that @digital-wonderland provided.
Thanks @kenisteward !

**Comment by kenisteward** [comment]
No problem. Thanks for making the loader. I hate that I can't use postcss- loader to do this but so be it
